io/net/tcpdump.py

```diff
@@ -63,8 +86,18 @@ def test(self):
         obj = process.SubProcess(cmd, verbose=False, shell=True)
         obj.start()
         output_file = os.path.join(self.outputdir, 'tcpdump')
-        cmd = "tcpdump -i %s -n -c %s -w '%s'" % (self.iface, self.count,
-                                                  output_file)
+        cmd = "tcpdump -i %s -n -c %s" % (self.iface, self.count)
+        if self.option in ('host', 'src'):
+            cmd = "%s %s %s" % (cmd, self.option, self.host_ip)
+        elif self.option == "dst":
+            cmd = "%s %s %s" % (cmd, self.option, self.peer_ip)
+        elif self.option in ('tcp', 'udp', 'icmp'):
+            obj = self.nping(self.option)
+            obj.start()
+            cmd = "%s %s" % (cmd, self.option)
+        else:
+            cmd = "%s %s" % (cmd, self.option)
+        cmd = "%s -w '%s'" % (cmd, output_file)
```

Comment:
This can be enhanced further to have `cmd` defined as a generic variable and then options added based on yaml parameters.